Add documentation and devcontainer for pipeline prototyping

AlexVCaron opened the pull request:
## Type of improvement

**If submitting a new module or fixing a bug, please use the appropriate template.**

- [x] Documentation
- [ ] Development tools (e.g. linter, formatter, etc.)
- [x] Development container
- [ ] Global update (please specify)
- [ ] Other (please specify)

## Describe your improvement

The title says it all. New pages of documentation linked in the readme. New containers as well, check them out when reopening your environment in a container now !

## Describe how to test your improvement

- Check the documentation for spelling
- Test the links
- Check the TOC
- Test the devcontainers

## Checklist before requesting a review

- [ ] Ensure the syntax is correct (**EditorConfig** and **Prettier** must pass)
- [ ] Run the test suites if your changes affect any module
- [ ] Regenerate the **Poetry** lock file if you have updated the dependencies
- [ ] Ensure the documentation is up-to-date

arnaudbore commented on README.md:
````diff
@@ -106,11 +131,126 @@ export NFCORE_SUBWORKFLOWS_GIT_REMOTE=https://github.com/scilus/nf-neuro.git
 
-# Developing in `nf-neuro`
+```bash
+nf-core modules info denoising/nlmeans
````
missing nf-core.yml and pyproject.toml

AlexVCaron commented:
@gagnonanthony @arnaudbore I updated the documentation, new structure that lightens the README. I also added the content @gagnonanthony created in a PR on my repository directly, since there is a section dedicated to it in this version. Tell me what you think. I believe it's on a good path, but there is still some massaging to do and visuals to include.
